src/dask_awkward/layers/layers.py

```diff
@@ -141,6 +141,11 @@ def project_columns(self, columns: list[str]) -> AwkwardInputLayer:
             if len(columns) == 0:
                 columns = self._meta.fields[:1]
 
+            # requested columns to original order; adds on extra columns, which are probably
+            # wildcard ones
+            columns = [c for c in self.columns if c in columns] + [
+                c for c in columns if c not in self.columns
+            ]
```

Comment:
`self.columns` can be `None` (probably usually is) so I think we need to go through grabbing columns from the form.
```suggestion
            # requested columns to original order; adds on extra columns, which are probably
            # wildcard ones
            form_columns = self._meta.layout.form.columns()
            original = [c for c in form_columns  if c in columns]
            new = [c for c in columns if c not in form_columns]
            columns = new + original
```

Reply:
aha, ok. Are you OK with adding in "new" columns like this? I think they'll arise with num() applied to parquet, which produces columns like "field.*". I had these after the originals, you have them before - this should not matter.

Reply:
Ah yea I see, probably best to make it `original + new`